### ISSUE TITLE
Hardcoded probably wrong debuginfo handling

### ISSUE THREAD

**Issue by rohanpm** [release-engineering/ubi-population-tool#25]
ubipop has hardcoded handling of debuginfo RPMs here, where it expects that any given RPM has a corresponding "-debuginfo" RPM:

```python
    def _create_debuginfo_output_set(self):
        """
        Creates output set for debug repo. Content is based on current rpms output set.
        """
        packages = chain.from_iterable(self.out_repo_set.packages.values())
        for package in packages:
            name, ver, rel, _, arch = splitFilename(package.filename)
            debug_pkg_filename = "{n}-debuginfo-{v}-{r}.{a}.rpm".format(n=name,
                                                                        v=ver,
                                                                        r=rel,
                                                                        a=arch)
            self.out_repo_set.debug_rpms.append(Package(name, debug_pkg_filename))

        blacklisted = self.get_blacklisted_packages(self.out_repo_set.debug_rpms)
        self._diff_packages_by_filename(self.out_repo_set.debug_rpms, blacklisted)
```

A couple of points about this:

- Is that actually needed?  Can't these be pulled in with the same whitelist mechanism as used for binary RPMs?  This part isn't necessarily a bug, as long as output is as it should be then it doesn't really matter whether this comes from the whitelist or other means.
- However, debuginfo RPMs aren't the only kind of RPMs going to debuginfo repos.  At least the logic above is going to miss debugsource RPMs.  This certainly seems like a bug as there is no particular reason debuginfo should be included while debugsource should be skipped.  See https://bugzilla.redhat.com/show_bug.cgi?id=1666366 for a bit of info about debugsource RPMs and an example of where they'll be shipped.

**Comment by rohanpm** [comment]
OK, I'm pretty sure that hardcoding just debuginfo is indeed broken as some debuginfo packages can depend on others.

For example, have a look at build glibc-2.17-287.el7.  There's a glibc-debuginfo-2.17-287.el7 which depends on glibc-debuginfo-common-2.17-287.el7.  Hardcoding inclusion of only glibc-debuginfo would result in an uninstallable package.

**Comment by rbikar** [comment]
@rohanpm 
1) Thanks for pointing out these debugsource rpms.
2) I'll use the same whitelist mechanism as used for binary RPMs so we won't miss any whitelisted package - ubi config includes also debuginfo/debugsource packages
3) For source rpms, I have similar hardcoded handling now that maps every output binary rpm to its src.rpm counterpart. Is this ok or should I use the same whitelist mechanism as for binary rpms? I'm counting on that every binary rpm must have its src.rpm counterpart.

**Comment by rohanpm** [comment]
@rbikar yeah, the SRPM handling does look a bit wrong too.

```python
            if package.sourcerpm_filename is None:
                name, ver, rel, _, _ = splitFilename(package.filename)
                package.sourcerpm_filename = "{n}-{v}-{r}.src.rpm".format(n=name, v=ver, r=rel)
```

You can't take name of a binary package to get the name of a source RPM.  The binary packages are using subpackage names which can be different from the top-level package.  A simple example is that one SRPM might be named python-foo and might create python2-foo, python3-foo subpackages.

Pulp should know what's the correct SRPM for any given binary RPM.  I think that should be used and probably not whitelist.

**Comment by rbikar** [comment]
Fixed by #36
